### ISSUE TITLE
Add support for default_value_expression in SchemaField

### ISSUE THREAD

**Issue by JoaoPaes-at-Dynamox** [googleapis/python-bigquery#1398]
Thanks for stopping by to let us know something could be better!

**PLEASE READ**: If you have a support contract with Google, please create an issue in the [support console](https://cloud.google.com/support/) instead of filing on GitHub. This will ensure a timely response.

 **Is your feature request related to a problem? Please describe.**
I'm developing a wrapper library to use `pydantic.BaseModel` as means to define the `TableSchema`. As part of this wrapper, I would like to provide "server default" values, such as `CURRENT_DATETIME`, `CURRENT_TIMESTAMP`, so on, to `datetime.datetime` fields;

 **Describe the solution you'd like**

```python
client = bigquery.Client("someproject")

bq_table_schema_field = SchemaField(..., default_value_expression="CURRENT_DATETIME()")

table_with_default_value = Table("my_table", schema=[schema_field])

client.create_table(table_with_default_value)
```

The field `"defaultValueExpression"` is a valid [REST API field](https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#TableFieldSchema) and would be a direct mapping from the initializer argument `default_value_expression` to the API `"defaultValueExpression"` field.

 **Describe alternatives you've considered**

The alternative to this, would be construct a DDL statement for creating tables which columns can have default values via `DEFAULT` keyword

```python
base_create = f"CREATE TABLE {somedataset_name}.{table_with_default_value.name} ("
for field in fields:
    field_as_column_statement = f'{field.name} {as_bq_type(field.type)} {"DEFAULT" if field.has_server_default else ""} {field.server_default_statement if field.has_server_default else ""}, '
    base_create = base_create + field_as_column_statement
base_create = base_create + ");"
```

Which is very cumbersome and ugly, and useless, since the REST API already provides a field for default value expressions.

**Comment by JoaoPaes-at-Dynamox** [comment]
Actually, a very much simpler, but extremely bad practice, way of solving this is

```python
...

bq_table_schema_field = SchemaField(...)
bq_table_schema_field._properties["defaultValueExpression"] =  "CURRENT_DATETIME()"

...
```

Which is what I'm currently using.

**Comment by JoaoPaes-at-Dynamox** [comment]
Just figured out that you included the parameter within the `3.4` release. I'm closing this.

**Comment by tswast** [comment]
Closed by https://github.com/googleapis/python-bigquery/pull/1408
